### ISSUE TITLE
Error creating dummy predictions: {'error': 'Result queue is empty', 'configuration_origin': 'DUMMY'} 

### ISSUE THREAD

**Issue by iXanthos** [automl/auto-sklearn#772]
Greetings,

I am trying to run `autosklearn 0.6.0` on an HPC infrastructure. I am using the `kr-vs-kp` dataset (https://www.openml.org/d/3) and I have made all the values of the dataset numeric. The error I get is:

`Unhandled exception in thread started by 
libgcc_s.so.1 must be installed for pthread_cancel to work
Traceback (most recent call last):
  File "main_script.py", line 41, in <module>
    autosklearn_script.autosklearn_func(train_data,train_target,test_data,test_target,dataset_info_dict,task,time_threshold)
  File "/users/pr004/ixanthos/test_files/autosklearn_script.py", line 31, in autosklearn_func
    automl.fit(train_data, train_target, metric=autosklearn.metrics.roc_auc)
  File "/apps/applications/pytorch/1.3.1/lib/python3.6/site-packages/autosklearn/estimators.py", line 664, in fit
    dataset_name=dataset_name,
  File "/apps/applications/pytorch/1.3.1/lib/python3.6/site-packages/autosklearn/estimators.py", line 337, in fit
    self._automl[0].fit(**kwargs)
  File "/apps/applications/pytorch/1.3.1/lib/python3.6/site-packages/autosklearn/automl.py", line 996, in fit
    load_models=load_models,
  File "/apps/applications/pytorch/1.3.1/lib/python3.6/site-packages/autosklearn/automl.py", line 208, in fit
    only_return_configuration_space=only_return_configuration_space,
  File "/apps/applications/pytorch/1.3.1/lib/python3.6/site-packages/autosklearn/automl.py", line 384, in _fit
    num_run = self._do_dummy_prediction(datamanager, num_run)
  File "/apps/applications/pytorch/1.3.1/lib/python3.6/site-packages/autosklearn/automl.py", line 313, in _do_dummy_prediction
    raise ValueError("Dummy prediction failed: %s " % str(additional_info))
ValueError: Dummy prediction failed: {'error': 'Result queue is empty', 'configuration_origin': 'DUMMY'} `

Any ideas?

Thank you in advance.

**Comment by willeppy** [comment]
I'm having a similar issue. I'm running this from a Jupyter Notebook and it seems to arise when I try to create a second classifier. For example, if I make a new classifier from `autosklearn.classification.AutoSklearnClassifier()` it works fine the first time but if I try to make another classifier later in the same notebook I am getting this error.

**Comment by iXanthos** [comment]
> I'm having a similar issue. I'm running this from a Jupyter Notebook and it seems to arise when I try to create a second classifier. For example, if I make a new classifier from `autosklearn.classification.AutoSklearnClassifier()` it works fine the first time but if I try to make another classifier later in the same notebook I am getting this error.

Is there a way to replicate this? Because I cannot in my Jupyter environment. I have trouble replicating the error I came across myself, so I am trying to understand when it occurs.

**Comment by mfeurer** [comment]
@iXanthos does this issue still appear with the latest release of Auto-sklearn (0.7.0)?

**Comment by arnaldog12** [comment]
yes, @mfeurer. I have just tested.

**Comment by mfeurer** [comment]
@arnaldog12 do you also get the error message: 

> Unhandled exception in thread started by libgcc_s.so.1 must be installed for pthread_cancel to work

**Comment by rabsr** [comment]
I am also having similar issue. However I don't get the following error message:

> Unhandled exception in thread started by libgcc_s.so.1 must be installed for pthread_cancel to work

**Comment by rabsr** [comment]
I was getting this error due to insufficient memory. The problem solved for me once I increased ml_memory_limt.

**Comment by mfeurer** [comment]
That's interesting, thanks for the pointer @rabsr 

Does the error look like, i.e. does it contain the threading memory error?
```
Unhandled exception in thread started by <bound method Thread._bootstrap of <Thread(QueueFeederThread, initial daemon)>>
Traceback (most recent call last):
  File "/home/feurerm/miniconda/3-4.5.4/envs/autosklearn/lib/python3.7/threading.py", line 885, in _bootstrap
MemoryError
[ERROR] [2020-08-14 11:15:37,969:AutoML(1):breast_cancer] Error creating dummy predictions: {'error': 'Result queue is empty', 'configuration_origin': 'DUMMY'} 
Traceback (most recent call last):
  File "/home/feurerm/sync_dir/projects/automl_competition_2015/auto-sklearn/examples/example_holdout.py", line 45, in <module>
    automl.fit(X_train, y_train, dataset_name='breast_cancer')
  File "/home/feurerm/sync_dir/projects/automl_competition_2015/auto-sklearn/autosklearn/estimators.py", line 686, in fit
    dataset_name=dataset_name,
  File "/home/feurerm/sync_dir/projects/automl_competition_2015/auto-sklearn/autosklearn/estimators.py", line 374, in fit
    self._automl[0].fit(**kwargs)
  File "/home/feurerm/sync_dir/projects/automl_competition_2015/auto-sklearn/autosklearn/automl.py", line 1072, in fit
    load_models=load_models,
  File "/home/feurerm/sync_dir/projects/automl_competition_2015/auto-sklearn/autosklearn/automl.py", line 398, in fit
    num_run = self._do_dummy_prediction(datamanager, num_run)
  File "/home/feurerm/sync_dir/projects/automl_competition_2015/auto-sklearn/autosklearn/automl.py", line 277, in _do_dummy_prediction
    % (str(status), str(additional_info)))
ValueError: Dummy prediction failed with run state StatusType.CRASHED and additional output: {'error': 'Result queue is empty', 'configuration_origin': 'DUMMY'}.

Process finished with exit code 1
```

**Comment by rabsr** [comment]
@mfeurer  I was running in Jupyter Notebook and was using Spark for parallel execution. 
The error thrown by spark job:

```  File "<ipython-input-8-6d22906a7950>", line 2, in <lambda>
  File "<ipython-input-6-0b9e1875cdb0>", line 24, in get_spawn_classifier
  File “/tmp/spark/userFiles/autosklearn/estimators.py", line 681, in fit
    dataset_name=dataset_name,
  File "/autosklearn/estimators.py", line 351, in fit
    self._automl[0].fit(**kwargs)
  File "/tmp/spark/user-files/autosklearn/automl.py", line 1026, in fit
    load_models=load_models,
  File "/tmp/spark/user-files/autosklearn/automl.py", line 213, in fit
    only_return_configuration_space=only_return_configuration_space,
  File "/tmp/spark/user-files/autosklearn/automl.py", line 397, in _fit
    num_run = self._do_dummy_prediction(datamanager, num_run)
  File "/tmp/spark/user-files/autosklearn/automl.py", line 318, in _do_dummy_prediction
    raise ValueError("Dummy prediction failed: %s " % str(additional_info))
ValueError: Dummy prediction failed: {'configuration_origin': 'DUMMY', 'error': 'Result queue is empty'} ```

**Comment by mfeurer** [comment]
Thanks @rabsr. Is this stdout/stderr or what is given in a log file? Unfortunately, we currently don't have a nice way to figuring out what goes wrong with the pynisher here, so we need to work on that. I'll ask around with my colleagues who had interest in improving the pynisher as well and see if we can get something started.

**Comment by rabsr** [comment]
@mfeurer This is from stdout/stderr.

**Comment by mfeurer** [comment]
Dear users who had this issue, since Auto-sklearn 0.11.1 the output of a failed DummyClassifier should be way more informative. Is one of you able to reproduce this issue and can post the error message here? Thanks a lot!

**Comment by zgm0407** [comment]
Auto-sklearn 0.12.0
Same code which was okay at another terminal. 


extra_trees_preproc_for_classification
adaboost
0
[ERROR] [2021-01-22 20:36:11,483:AutoML(1):3e813b19775cc3e8ac51e52b04d97ca2] Dummy prediction failed with run state StatusType.MEMOUT and additional output: {'error': 'Memout (used more than 3072 MB).', 'configuration_origin': 'DUMMY'}.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-13-93e07b525b9c> in <module>
     37                 resampling_strategy_arguments={'folders':5}
     38             )
---> 39             automl.fit(X_train,y_train,X_test,y_test)
     40 
     41 

~/.local/lib/python3.6/site-packages/autosklearn/estimators.py in fit(self, X, y, X_test, y_test, feat_type, dataset_name)
    595             y_test=y_test,
    596             feat_type=feat_type,
--> 597             dataset_name=dataset_name,
    598         )
    599 

~/.local/lib/python3.6/site-packages/autosklearn/estimators.py in fit(self, **kwargs)
    354             output_folder=self.output_folder,
    355         )
--> 356         self.automl_.fit(load_models=self._load_models, **kwargs)
    357 
    358         return self

~/.local/lib/python3.6/site-packages/autosklearn/automl.py in fit(self, X, y, X_test, y_test, feat_type, dataset_name, only_return_configuration_space, load_models)
   1320             dataset_name=dataset_name,
   1321             only_return_configuration_space=only_return_configuration_space,
-> 1322             load_models=load_models,
   1323         )
   1324 

~/.local/lib/python3.6/site-packages/autosklearn/automl.py in fit(self, X, y, task, X_test, y_test, feat_type, dataset_name, only_return_configuration_space, load_models)
    584         # == Perform dummy predictions
    585         num_run = 1
--> 586         self._do_dummy_prediction(datamanager, num_run)
    587 
    588         # = Create a searchspace

~/.local/lib/python3.6/site-packages/autosklearn/automl.py in _do_dummy_prediction(self, datamanager, num_run)
    414                 raise ValueError(
    415                     "Dummy prediction failed with run state %s and additional output: %s."
--> 416                     % (str(status), str(additional_info))
    417                 )
    418 

ValueError: Dummy prediction failed with run state StatusType.MEMOUT and additional output: {'error': 'Memout (used more than 3072 MB).', 'configuration_origin': 'DUMMY'}.

**Comment by MaxHeuillet** [comment]
Hello I am also having this issue...

I run Autosklearn2 on a DGX on a small sized file, so it can not be linked to a memory error.

`[ERROR] [2021-02-04 15:00:26,097:Client-AutoML(1):d8cda2dc35a43ca16e484b4f0d0ed92b] Dummy prediction failed with run state StatusType.CRASHED and additional output: {'error': 'Result queue is empty', 'exit_status': "<class 'pynisher.limit_function_call.AnythingException'>", 'subprocess_stdout': '', 'subprocess_stderr': '', 'exitcode': 0, 'configuration_origin': 'DUMMY'}.`

The parameters I have are: 

`automl = autosklearn.classification.AutoSklearnClassifier( ensemble_size=1, time_left_for_this_task = 30 )
 automl.fit(train_x, train_y)`

What I don't understand is that it works on my local laptop but not on the DGX.

Please help, I need your framework 2.0 working for my experiments. 

Thanks !

**Comment by mfeurer** [comment]
Sorry @MaxHeuillet I must have missed your comment. Could you please let me know what version of Auto-sklearn you're using and also check whether it works if you use a longer `time_left_for_this_task`?

**Comment by sofidenner** [comment]
Hi! A possibly related issue: with `auto-sklearn==0.13.0`, I run exactly the same code, with the same dataset in three different virtual machines:
- in a vm with 4 cores and 15Gb of RAM: works ok ✅
- in a vm with 8 cores and 30Gb of RAM: works ok ✅
- in a vm with 40 cores and 157 Gb of RAM: fails ❌  with the following error:

`
ValueError: Dummy prediction failed with run state StatusType.CRASHED and additional output: {'error': 'Result queue is empty', 'exit_status': "<class 'pynisher.limit_function_call.AnythingException'>", 'subprocess_stdout': '', 'subprocess_stderr': 'Process pynisher function call:\nTraceback (most recent call last):\n  File "/usr/local/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap\n    self.run()\n  File "/usr/local/lib/python3.7/multiprocessing/process.py", line 99, in run\n    self._target(*self._args, **self._kwargs)\n  File "/usr/local/lib/python3.7/site-packages/pynisher/limit_function_call.py", line 133, in subprocess_func\n    return_value = ((func(*args, **kwargs), 0))\n  File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/__init__.py", line 40, in fit_predict_try_except_decorator\n    return ta(queue=queue, **kwargs)\n  File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/train_evaluator.py", line 1164, in eval_holdout\n    budget_type=budget_type,\n  File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/train_evaluator.py", line 194, in __init__\n    budget_type=budget_type,\n  File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/abstract_evaluator.py", line 199, in __init__\n    threadpool_limits(limits=1)\n  File "/usr/local/lib/python3.7/site-packages/threadpoolctl.py", line 171, in __init__\n    self._original_info = self._set_threadpool_limits()\n  File "/usr/local/lib/python3.7/site-packages/threadpoolctl.py", line 280, in _set_threadpool_limits\n    module.set_num_threads(num_threads)\n  File "/usr/local/lib/python3.7/site-packages/threadpoolctl.py", line 659, in set_num_threads\n    return set_func(num_threads)\nKeyboardInterrupt\n', 'exitcode': 1, 'configuration_origin': 'DUMMY'}.
`

This is the code I was running:
```python
automl = AutoSklearnClassifier(time_left_for_this_task=600, metric=roc_auc)
automl.fit(x_train, y_train, x_validation, y_validation)
```

Limiting the number of cores with the param `nproc` seems to work, but it's a pity that we cannot take advantage of larger infra :(

**Comment by mfeurer** [comment]
Hey, this issue is getting hard to follow through as we have changed Auto-sklearn quite a bit since most answers. 

@MaxHeuillet: if this is still relevant for you could you please open a new issue as your issue appears to be related to a special hardware platform. Not sure if we can help, but it'll make it easier to keep track of things.
@rabsr: We have improved the printed output quite a bit. Could you please open a new issue if you are still experiencing this?
@sofidenner: As your issue already has extra output that we have added quite some time ago it would be great if you could open a new issue with what you already posted and also a) whether it happens repeatedly or only stochastically, b) additional info on the dataset you're using.

**Comment by ajaynarayanan** [comment]
> Hi! A possibly related issue: with `auto-sklearn==0.13.0`, I run exactly the same code, with the same dataset in three different virtual machines:
> 
> * in a vm with 4 cores and 15Gb of RAM: works ok ✅
> * in a vm with 8 cores and 30Gb of RAM: works ok ✅
> * in a vm with 40 cores and 157 Gb of RAM: fails ❌  with the following error:
> 
> `ValueError: Dummy prediction failed with run state StatusType.CRASHED and additional output: {'error': 'Result queue is empty', 'exit_status': "<class 'pynisher.limit_function_call.AnythingException'>", 'subprocess_stdout': '', 'subprocess_stderr': 'Process pynisher function call:\nTraceback (most recent call last):\n File "/usr/local/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap\n self.run()\n File "/usr/local/lib/python3.7/multiprocessing/process.py", line 99, in run\n self._target(*self._args, **self._kwargs)\n File "/usr/local/lib/python3.7/site-packages/pynisher/limit_function_call.py", line 133, in subprocess_func\n return_value = ((func(*args, **kwargs), 0))\n File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/__init__.py", line 40, in fit_predict_try_except_decorator\n return ta(queue=queue, **kwargs)\n File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/train_evaluator.py", line 1164, in eval_holdout\n budget_type=budget_type,\n File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/train_evaluator.py", line 194, in __init__\n budget_type=budget_type,\n File "/usr/local/lib/python3.7/site-packages/autosklearn/evaluation/abstract_evaluator.py", line 199, in __init__\n threadpool_limits(limits=1)\n File "/usr/local/lib/python3.7/site-packages/threadpoolctl.py", line 171, in __init__\n self._original_info = self._set_threadpool_limits()\n File "/usr/local/lib/python3.7/site-packages/threadpoolctl.py", line 280, in _set_threadpool_limits\n module.set_num_threads(num_threads)\n File "/usr/local/lib/python3.7/site-packages/threadpoolctl.py", line 659, in set_num_threads\n return set_func(num_threads)\nKeyboardInterrupt\n', 'exitcode': 1, 'configuration_origin': 'DUMMY'}.`
> 
> This is the code I was running:
> 
> ```python
> automl = AutoSklearnClassifier(time_left_for_this_task=600, metric=roc_auc)
> automl.fit(x_train, y_train, x_validation, y_validation)
> ```
> 
> Limiting the number of cores with the param `nproc` seems to work, but it's a pity that we cannot take advantage of larger infra :(

@sofidenner: Thanks for the suggestion on limiting the `nproc`. I faced the similar issue while running on docker with 12 cpu processors machine. I was able to get rid of this issue by allocating only 11 cpu cores to the docker image.

**Comment by sofidenner** [comment]
Thanks @mfeurer , I've opened this new issue https://github.com/automl/auto-sklearn/issues/1236 with the info you request.
